src/dex-ui/pages/dao/DAOProposals/Forms/DAOTokenTransferDetailsForm.tsx

```diff
@@ -13,44 +15,66 @@ export interface TokenTransferLocationState {
 }
 
 export function DAOTokenTransferDetailsForm() {
-  /*
-   * TODO: Replace Assets Input with Dropdown
-   * const { wallet } = useDexContext(({ wallet }) => ({ wallet }));
-   * const accountTokenBalancesQueryResults = useAccountTokenBalances(wallet.savedPairingData?.accountIds[0] ?? "");
-   * const { data: tokenBalances } = accountTokenBalancesQueryResults;
-   */
   const { state } = useLocation() as TokenTransferLocationState;
   const { safeAccountId, daoType, proposalType } = useOutletContext<CreateDAOProposalContext>();
   const {
+    getValues,
     setValue,
     register,
+    watch,
     formState: { errors },
   } = useFormContext<CreateDAOTokenTransferForm>();
+  watch("tokenId");
+  const accountTokenBalancesQueryResults = useAccountTokenBalances(safeAccountId ?? "");
+  const { data: tokenBalances } = accountTokenBalancesQueryResults;
+  const selectedAsset = tokenBalances?.find((asset: TokenBalance) => asset.tokenId === getValues().tokenId);
 
   if (proposalType !== DAOProposalType.TokenTransfer) {
     setValue("type", DAOProposalType.TokenTransfer);
   }
 
-  /*
-   * TODO: Replace Assets Input with Dropdown
-   * const assetDropdownOptions =
-   *   tokenBalances
-   *     ?.filter((asset: TokenBalance) => asset.symbol !== "ℏ")
-   *     ?.map((asset: TokenBalance) => {
-   *       const { symbol, tokenId } = asset;
-   *       return {
-   *         label: symbol,
-   *         value: tokenId,
-   *       };
-   *      }) ?? [];
-   */
+  const validateAmount = (value: number) => {
+    if (value <= 0) {
+      return "Amount must be greater than 0.";
+    }
+    const tokenId = getValues().tokenId;
+    if (!tokenId) {
+      return "Token must be selected.";
+    }
+    const selectedAsset = tokenBalances?.find((asset: TokenBalance) => asset.tokenId === tokenId);
+    if (!selectedAsset?.balance || selectedAsset?.balance <= 0) {
+      return "Token balance must be greater than 0.";
+    }
+    if (value > selectedAsset?.balance) {
+      return "Amount must be less than or equal to token balance.";
+    }
+    return true;
+  };
+
+  const assetDropdownOptions =
+    tokenBalances
+      ?.filter((asset: TokenBalance) => asset.symbol !== "ℏ")
+      ?.filter((asset: TokenBalance) => asset.balance > 0)
+      ?.map((asset: TokenBalance) => {
+        const { symbol, tokenId } = asset;
+        return {
+          label: symbol,
+          value: tokenId,
+        };
+      }) ?? [];
 
   useEffect(() => {
     if (state?.tokenId) {
       setValue("tokenId", state?.tokenId);
     }
   }, [setValue, state?.tokenId]);
 
+  useEffect(() => {
+    if (daoType === DAOType.MultiSig) {
```

Comment:
DAOType.MultiSig is equal to "MultiSig". daoType is "multisig" when the DAO is a MultiSig DAO. Therefore, this clause will never be true.

Reply:
You are right. Made the changes are per your suggestions.